pkg/util/slices.go

```diff
@@ -48,3 +51,57 @@ func CombineUnique[T constraints.Ordered](a, b []T) []T {
 	}
 	return a
 }
+
+// Filter removes all elements from collection that do not satisfy predicate.
+// NOTE: Collection is modified in place. To create a new slice, utilize
```

Comment:
Drive-by: I would NEVER expect a `Filter()` function to modify a collection in-place. To me, this is asking for bugs. Up to you whether you want this behavior specifically for some reason (performance?)

Reply:
Yeah, I was thinking that it could be used in performance critical cases if we did the filter in place. I started to use this function elsewhere and I 100% agree with you.

I'll update it as soon as my working tree is clean 👍

Reply:
(Something to consider: a function name like `FilterInPlace()` would make clear the intent and avoid the possibility of bugs.)

Reply:
For the in place case, I think we'd want to fallback to utilizing a for loop? Or we can add `FilterInPlace` once it makes sense to do so :D

Reply:
I agree that if you want the in-place behavior (due to performance probably), you likely want to be writing a for-loop, taking advantage of function call inlining, etc.

Reply:
Fixed!